tcms/issuetracker/tests/test_azureboards.py

```diff
@@ -0,0 +1,201 @@
+
+        # verify that LR has been added to TE
+        self.assertTrue(LinkReference.objects.filter(
+            execution=self.execution_1,
+            url=result['response'],
+            is_defect=True,
+        ).exists())
+
+        # Set work item done after test is finished
+        close_issue(self.integration.rpc, new_issue_id)
+
+    def test_report_issue_from_test_execution_empty_baseurl_exception(self):
+        # simulate user clicking the 'Report bug' button in TE widget, TR page
+        bug_system = BugSystem.objects.create(  # nosec:B106:hardcoded_password_funcarg
+            name='Azure for kiwitcms/test-azure-integration Exception',
+            tracker_type='tcms.issuetracker.azure_boards.AzureBoards',
+            base_url='incorrect_url',
+            api_password=os.getenv('AZURE_BOARDS_INTEGRATION_API_TOKEN'),
+        )
+        integration = AzureBoards(bug_system, None)
+        result = self.rpc_client.Bug.report(self.execution_1.pk, integration.bug_system.pk)
+        self.assertIn('_workitems/create/Issue', result['response'])
+
+
+@unittest.skipUnless(os.getenv('TEST_BUGTRACKER_INTEGRATION'),
+                     'Bug tracker integration testing not enabled')
+class TestAzureAPI(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        api_password = os.getenv('AZURE_BOARDS_INTEGRATION_API_TOKEN')
+        base_url = 'https://dev.azure.com/kiwitcms/boards-integration'
+        cls.api_instance = AzureAPI(base_url, api_password)
+
+        test_create_body = [
+            {
+                "op": "add",
+                "path": "/fields/System.Title",
+                "from": "null",
+                "value": "Test Issue"
+            }]
+        result = cls.api_instance.create_issue(test_create_body)
+        cls.issue_id = result['id']
+
+    @classmethod
+    def tearDownClass(cls):
+        close_issue(cls.api_instance, cls.issue_id)
+        return super().tearDownClass()
+
+    def test_create_issue(self):
+        test_create_body = [
+            {
+                "op": "add",
+                "path": "/fields/System.Title",
+                "from": "null",
+                "value": "Test Issue"
+            }]
+        result = self.api_instance.create_issue(test_create_body)
+        self.assertEqual(result['fields']['System.Title'], "Test Issue")
+
+        # Close work item after test is finished.
+        close_issue(self.api_instance, result['id'])
+
+    def test_update_issue(self):
+        test_update_body = [
+            {
+                "op": "replace",
+                "path": "/fields/System.Description",
+                "from": "null",
+                "value": "Updated Description"
+            }]
+        result = self.api_instance.update_issue(self.issue_id, test_update_body)
+        self.assertEqual(result['fields']['System.Description'], "Updated Description")
```

Comment:
This breaks the integration tests:
https://github.com/kiwitcms/Kiwi/pull/2036/checks?check_run_id=1326143291

```
2020-10-29T11:10:21.8015735Z Traceback (most recent call last):
2020-10-29T11:10:21.8016124Z     Running deferred SQL...
2020-10-29T11:10:21.8018150Z   File "/home/runner/work/Kiwi/Kiwi/tcms/issuetracker/tests/test_azureboards.py", line 56, in test_details
2020-10-29T11:10:21.8018750Z Running migrations:
2020-10-29T11:10:21.8020199Z     result['description'])
2020-10-29T11:10:21.8020649Z   Applying contenttypes.0001_initial... OK
2020-10-29T11:10:21.8022360Z AssertionError: 'The whole boards-integration project is public so this issue should be public as well' not found in 'Updated Description'
```

Use the issue_id from setup instead of the pre-existing one.

Reply:
> This breaks the integration tests:
> https://github.com/kiwitcms/Kiwi/pull/2036/checks?check_run_id=1326143291
> 
> ```
> 2020-10-29T11:10:21.8015735Z Traceback (most recent call last):
> 2020-10-29T11:10:21.8016124Z     Running deferred SQL...
> 2020-10-29T11:10:21.8018150Z   File "/home/runner/work/Kiwi/Kiwi/tcms/issuetracker/tests/test_azureboards.py", line 56, in test_details
> 2020-10-29T11:10:21.8018750Z Running migrations:
> 2020-10-29T11:10:21.8020199Z     result['description'])
> 2020-10-29T11:10:21.8020649Z   Applying contenttypes.0001_initial... OK
> 2020-10-29T11:10:21.8022360Z AssertionError: 'The whole boards-integration project is public so this issue should be public as well' not found in 'Updated Description'
> ```
> 
> Use the issue_id from setup instead of the pre-existing one.

I would not expect it to fail to be honest, i couldn't get why its failing.
It has nothing to do with the issues created during the tests in `class TestAzureAPI(unittest.TestCase):`

Reply:
That would be great if you can somehow provide me the api_key for https://dev.azure.com/kiwitcms/boards-integration otherwise i can't debug it.

Reply:
Token sent. 

> I would not expect it to fail, to be honest i couldn't get why its failing.

I see now that setup is trying to create a new issue and then passed the ID around. Probably the creation fails without a traceback and a noticeable error and you end up with ID=1.

Reply:
No, actually `test_details` case only tests the content and title of;
https://dev.azure.com/kiwitcms/boards-integration/_workitems/edit/1/

It worked without fail when ı run it locally. Could you please make it run the checks again by merging it to `pr2034` branch. I suspect from something overriding test execution.

Reply:
IDK why it failed before. Looks good now https://github.com/kiwitcms/Kiwi/pull/2036/checks?check_run_id=1331299786